src/v/jl/metadata.h

```diff
@@ -0,0 +1,54 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#pragma once
+
+#include "serde/envelope.h"
+#include "utils/named_type.h"
+#include "utils/uuid.h"
+
+#include <__compare/common_comparison_category.h>
+#include <fmt/ostream.h>
+
+#include <ostream>
+#include <stdint.h>
+#include <tuple>
+
+namespace jumbo_log {
+using jsn = named_type<uint64_t, struct jumbo_segment_sequence_number_tag>;
+using write_intent_id_t = named_type<uint64_t, struct write_intent_id_tag>;
+
+struct segment_object {
+    static constexpr serde::version_t redpanda_serde_version = 0;
+    static constexpr serde::version_t redpanda_serde_compat_version = 0;
```

Comment:
can't use envelope because the same type is used in the rpc layer and that one is using adl

inheriting from the envelope makes reflection::arity return number of members + 1 something which adl reflection doesn't expect